R/pkg/R/SQLContext.R

```diff
@@ -214,18 +214,45 @@ setMethod("toDF", signature(x = "RDD"),
 #' sc <- sparkR.init()
 #' sqlContext <- sparkRSQL.init(sc)
 #' path <- "path/to/file.json"
-#' df <- jsonFile(sqlContext, path)
+#' df <- read.json(sqlContext, path)
 #' }
 
-jsonFile <- function(sqlContext, path) {
+read.json <- function(sqlContext, ...) {
   # Allow the user to have a more flexible definiton of the text file path
-  path <- suppressWarnings(normalizePath(path))
+  paths <- if (length(list(...)) > 1) {
+    lapply(list(...), function(x) suppressWarnings(normalizePath(x)))
+  } else {
+    as.list(suppressWarnings(normalizePath(splitString(...))))
+  }
   # Convert a string vector of paths to a string containing comma separated paths
-  path <- paste(path, collapse = ",")
-  sdf <- callJMethod(sqlContext, "jsonFile", path)
+  read <- callJMethod(sqlContext, "read")
+  sdf <- callJMethod(read, "json", paths)
   dataFrame(sdf)
 }
 
+#' Create a DataFrame from a JSON file.
```

Comment:
yes, see createDataFrame and as.DataFrame.
They should have the same `@rdname` but distinct `@name`